avocado/utils/genio.py

```diff
@@ -201,3 +202,25 @@ def write_file_or_fail(filename, data):
     except OSError as details:
         raise GenIOError("The write to %s failed: %s" % (
                          filename, details))
+
+
+def find_pattern(filename,  pattern):
+    """
+    Module for match pattern in a specified file
+
+    :param filename: Path to file
+    :type filename: str
+    :param pattern: pattern that need to match in file
+    :type pattern: str
+    :return: True when pattern matches in file if not
+             retun False
+    """
+    if not os.path.isfile(filename):
+        raise GenIOError('invalid file %s to match pattern %s'
+                         % (filename, pattern))
+    pattern = re.compile(r'%s' % pattern)
+
+    for line in open(filename, 'r').readlines():
```

Comment:
Have you considered, for your use cases, using `re.search(pattern, whole_file_content, re.MULTILINE)` ?